### PR TITLE
Add field in helmValues step to change releaseName

### PR DESCRIPTION
**Pull request by Rob0h**
What I Did
------------
Add field in helmValues step to change releaseName

How I Did it
------------
Serialize releaseName at resolveRelease step
Add field for UI in helmValues step to change releaseName

How to verify it
------------
- Tests (for serialization of releaseName)
- Run `ship init` against any helm chart, Change the releaseName at the helmValues step, Observe changes to rendered yaml

Description for the Changelog
------------
Add field in helmValues step to change releaseName

Picture of a Boat (not required but encouraged)
------------
🛶

### CODE REVIEW

File: pkg/api/spec.go [hunk 1]
```diff
@@ -1,5 +1,12 @@
 package api
 
+import (
+	"regexp"
+	"strings"
+)
+
+var releaseNameRegex = regexp.MustCompile("[^a-zA-Z0-9\\-]")
```
Comment (dexhorthy) [inline pkg/api/spec.go]: Its not clear from the name of this what its for -- it looks like it matches characters that are disallowed in a release name?

File: pkg/lifecycle/daemon/routes_v1.go [hunk 2]
```diff
@@ -122,12 +123,19 @@ func (d *V1Routes) saveHelmValues(c *gin.Context) {
...
-		debug.Log("event", "seralize.fail", "err", err)
+	debug.Log("event", "serialize.helmValues")
+	if err := d.StateManager.SerializeHelmValues(request.Values, string(chartDefaultValues)); err != nil {
+		debug.Log("event", "seralize.helmValues.fail", "err", err)
 		c.AbortWithError(http.StatusInternalServerError, errors.New("internal_server_error"))
 	}
+
+	debug.Log("event", "serialize.helmReleaseName")
+	if len(request.ReleaseName) > 0 {
+		if err := d.StateManager.SerializeReleaseName(request.ReleaseName); err != nil {
```
Comment (dexhorthy) [inline pkg/lifecycle/daemon/routes_v1.go]: why are we having the client set the release name?

File: pkg/api/spec.go (hunk 1 above)
Comment (dexhorthy) [reply pkg/api/spec.go]: I see that this is copy pasted, so I'm :shrug: on it.

File: pkg/lifecycle/render/helm/template_test.go [hunk 3]
```diff
@@ -82,7 +83,7 @@ func TestLocalTemplater(t *testing.T) {
 			expectedHelmValues: []string{
```
Comment (dexhorthy) [inline pkg/lifecycle/render/helm/template_test.go]: judging by the name of this test, I'd imagine its no longer required.

File: pkg/state/models.go [hunk 4]
```diff
@@ -14,6 +14,7 @@ type State interface {
 	CurrentKustomizeOverlay(filename string) (string, bool)
 	CurrentHelmValues() string
 	CurrentHelmValuesDefaults() string
+	CurrentReleaseName() string
```
Comment (dexhorthy) [inline pkg/state/models.go]: I don't think we need to continue adding "CurrentXXXX" methods to the `State` interface -- you should be able to just use `.Versioned()` and then access the release name directly?

File: pkg/lifecycle/daemon/routes_v1.go (hunk 2 above)
Comment (Rob0h) [reply pkg/lifecycle/daemon/routes_v1.go]: it's a client request 😅

**Pull request merged**